src/rsa.rs

```diff
@@ -151,7 +98,43 @@ rsa_pkcs1_padding!(RSA_PKCS1_SHA512, &digest::SHA512,
                    &SHA512_PKCS1_DIGESTINFO_PREFIX,
                    "Signing using RSA with PKCS#1 1.5 padding and SHA-512.");
 
-macro_rules! rsa_pkcs1 {
+rsa_pss_padding!(RSA_PSS_SHA1, &digest::SHA1,
+                   "Signing using RSA with PSS padding and SHA-1.", 0);
+rsa_pss_padding!(RSA_PSS_SHA256, &digest::SHA256,
+                   "Signing using RSA with PSS padding and SHA-256.", 0);
+rsa_pss_padding!(RSA_PSS_SHA384, &digest::SHA384,
+                   "Signing using RSA with PSS padding and SHA-384.", 0);
+rsa_pss_padding!(RSA_PSS_SHA512, &digest::SHA512,
+                   "Signing using RSA with PSS padding and SHA-512.", 0);
+
+rsa_pss_padding!(RSA_PSS_SHA1_20, &digest::SHA1,
```

Comment:
This is particularly tedious and not very flexible. Could make salt length an additional input to the encode/verify algorithms, but this is not something like to be shared by other padding schemes.

Alternatively, we enforce only a small number of options, e.g., salt length = digest length.

This current list is to cover all of the NIST tests in rsa_pss_{sign,verify}_tests.txt